torch/distributed/algorithms/_comm_hooks/default_hooks.py

```diff
@@ -0,0 +1,126 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import distributed_c10d
+
+
+class DefaultState(object):
```

Comment:
Great job to refactor into DefaultState and make LowPrecisionState from DefaultState!